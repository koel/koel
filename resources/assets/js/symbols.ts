import { DeepReadonly, InjectionKey, Ref } from 'vue'
import Overlay from '@/components/ui/Overlay.vue'
import DialogBox from '@/components/ui/DialogBox.vue'
import MessageToaster from '@/components/ui/MessageToaster.vue'
import Router from '@/router'

export type ReadonlyInjectionKey<T> = InjectionKey<[Readonly<T> | DeepReadonly<T>, Closure]>

export const RouterKey: InjectionKey<Router> = Symbol('Router')
export const OverlayKey: InjectionKey<Ref<InstanceType<typeof Overlay>>> = Symbol('Overlay')
export const DialogBoxKey: InjectionKey<Ref<InstanceType<typeof DialogBox>>> = Symbol('DialogBox')
export const MessageToasterKey: InjectionKey<Ref<InstanceType<typeof MessageToaster>>> = Symbol('MessageToaster')

export const SongsKey: ReadonlyInjectionKey<Ref<Song[]>> | InjectionKey<Ref<Song[]>> = Symbol('Songs')
export const CurrentSongKey: InjectionKey<Ref<Song | undefined>> = Symbol('CurrentSong')
export const SelectedSongsKey: ReadonlyInjectionKey<Ref<Song[]>> = Symbol('SelectedSongs')
export const SongListConfigKey: ReadonlyInjectionKey<Partial<SongListConfig>> = Symbol('SongListConfig')
export const SongListSortFieldKey: ReadonlyInjectionKey<Ref<SongListSortField>> = Symbol('SongListSortField')
export const SongListSortOrderKey: ReadonlyInjectionKey<Ref<SortOrder>> = Symbol('SongListSortOrder')
export const SongListFilterKeywordsKey: InjectionKey<Ref<string>> = Symbol('SongListFilterKeywords')

export const ModalContextKey: InjectionKey<Ref<Record<string, any>>> = Symbol('ModalContext')
