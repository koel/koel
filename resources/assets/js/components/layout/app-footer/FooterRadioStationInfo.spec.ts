import { expect, it } from 'vitest'
import { ref } from 'vue'
import factory from '@/__tests__/factory'
import UnitTestCase from '@/__tests__/UnitTestCase'
import { CurrentStreamableKey } from '@/symbols'
import Component from './FooterRadioStationInfo.vue'

new class extends UnitTestCase {
  protected test () {
    it('renders with current radio', () => {
      const station = factory('radio-station', {
        name: 'Classic Rock',
        logo: 'https://via.placeholder.com/150',
        description: 'The best classic rock hits',
        playback_state: 'Playing',
      })

      expect(this.render(Component, {
        global: {
          provide: {
            [<symbol>CurrentStreamableKey]: ref(station),
          },
        },
      }).html()).toMatchSnapshot()
    })
  }
}
