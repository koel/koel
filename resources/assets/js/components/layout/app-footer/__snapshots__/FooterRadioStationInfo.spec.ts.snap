// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`footerRadioStationInfo.vue > renders with current radio 1`] = `
<div data-v-5b109e0d="" class="playing station-info px-6 py-0 flex items-center content-start w-[84px] md:w-[420px] gap-5"><span data-v-5b109e0d="" class="logo block h-[55%] md:h-3/4 aspect-square rounded-full bg-cover"></span>
  <div data-v-5b109e0d="" class="meta overflow-hidden hidden md:block">
    <h3 data-v-5b109e0d="" class="title text-ellipsis overflow-hidden whitespace-nowrap">Classic Rock</h3>
    <p data-v-5b109e0d="" class="text-ellipsis overflow-hidden whitespace-nowrap">The best classic rock hits</p>
  </div>
</div>
`;
